app/jobs/set_catalog_record_ids_and_barcodes_csv_job.rb

```diff
@@ -16,77 +16,47 @@ class SetCatalogRecordIdsAndBarcodesCsvJob < GenericJob
   def perform(bulk_action_id, params)
     super
 
-    # Catalog record IDs, barcodes are nil if not selected for use.
-    update_druids, catalog_record_ids, barcodes, refresh = params_from(params)
+    with_csv_items(
```

Comment:
We can substantially reduce the size and complexity of this job by leaning on `GenericJob#with_csv_items` instead of replicating that method's logic here.